helix-view/src/document.rs

```diff
@@ -265,6 +276,174 @@ impl Document {
         }
     }
 
+    fn detect_indent_style(&mut self) {
+        // Determine whether a character is a line break.
+        //
+        // TODO: this is probably a generally useful utility function.  Where
+        // should we put it?
+        fn char_is_linebreak(c: char) -> bool {
```

Comment:
I think a submodule under `helix_core` could be used. Maybe `chars`? To match the `graphemes` submodule.

Reply:
Sounds good!  Maybe we'll also have a `string` module for `str` and friends?